go-controller/pkg/ovn/egressgw.go

```diff
@@ -0,0 +1,254 @@
+				_, stderr, err := util.RunOVNNbctl("--", "--may-exist", "--policy=src-ip", "--ecmp",
+					"lr-route-add", gr, podIP.IP+mask, gw.String())
+				if err != nil {
+					return fmt.Errorf("unable to add src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+				}
+				if nsInfo.podExternalRoutes[podIP.IP] == nil {
+					nsInfo.podExternalRoutes[podIP.IP] = make(map[string]string)
+				}
+				nsInfo.podExternalRoutes[podIP.IP][gw.String()] = gr
+			}
+		}
+	}
+	return nil
+}
+
+// deletePodExternalGW detects if a given pod is acting as an external GW and removes all routes in all namespaces
+// associated with that pod
+func (oc *Controller) deletePodExternalGW(pod *kapi.Pod) {
+	routingNamespaceAnnotation := pod.Annotations[routingNamespaceAnnotation]
+	if routingNamespaceAnnotation == "" {
+		return
+	}
+	klog.Infof("Deleting routes for external gateway pod: %s, for namespace(s) %s", pod.Name,
+		routingNamespaceAnnotation)
+	for _, namespace := range strings.Split(routingNamespaceAnnotation, ",") {
+		oc.deletePodGWRoutesForNamespace(pod.Name, namespace)
+	}
+}
+
+// deletePodGwRoutesForNamespace handles deleting all routes in a namespace for a specific pod GW
+func (oc *Controller) deletePodGWRoutesForNamespace(pod, namespace string) {
+	nsInfo := oc.getNamespaceLocked(namespace)
+	if nsInfo == nil {
+		return
+	}
+	defer nsInfo.Unlock()
+	// check if any gateways were stored for this pod
+	foundGws := nsInfo.routingExternalPodGWs[pod]
+	if foundGws == nil {
+		klog.Infof("No gateways found to remove for annotated gateway pod: %s on namespace: %s",
+			pod, namespace)
+		return
+	}
+
+	for _, gwIP := range foundGws {
+		// check for previously configured pod routes
+		for podIP, gwInfo := range nsInfo.podExternalRoutes {
+			if len(gwInfo) == 0 {
+				continue
+			}
+			gr := gwInfo[gwIP.String()]
+			if gr == "" {
+				continue
+			}
+			mask := GetIPFullMask(podIP)
+			// TODO (trozet): use the go bindings here and batch commands
+			_, stderr, err := util.RunOVNNbctl("--", "--if-exists", "--policy=src-ip",
+				"lr-route-del", gr, podIP+mask, gwIP.String())
+			if err != nil {
+				klog.Errorf("Unable to delete pod %s route to GR %s, GW: %s, stderr:%q, err:%v",
+					pod, gr, gwIP.String(), stderr, err)
+			} else {
+				klog.V(5).Infof("ECMP route deleted for pod: %s, on gr: %s, to gw: %s", pod,
+					gr, gwIP.String())
+				delete(nsInfo.podExternalRoutes[podIP], gwIP.String())
+				// clean up if there are no more routes for this podIP
+				if entry := nsInfo.podExternalRoutes[podIP]; len(entry) == 0 {
+					delete(nsInfo.podExternalRoutes, podIP)
+				}
+			}
+		}
+	}
+	delete(nsInfo.routingExternalPodGWs, pod)
+}
+
+// deleteGwRoutesForNamespace handles deleting all routes to gateways for a pod on a specific GR
+// This should only be called with a lock on nsInfo
+func (oc *Controller) deleteGWRoutesForNamespace(nsInfo *namespaceInfo) {
+	if nsInfo == nil {
+		return
+	}
+	// TODO(trozet): batch all of these with ebay bindings
```

Comment:
@trozet can't we batch them now? or would you rather wait.

Reply:
bindings are missing support for ecmp (and the auto next hop feature tbd). I'll open a PR to add it, but in the meantime we can hold off.

Reply:
https://github.com/eBay/go-ovn/blob/master/logical_router_static_route.go#L35

Reply:
As we discussed we will hold off on batching until we can get the bindings working.